index.js

```diff
@@ -0,0 +1,49 @@
+const buildRuntimeConfig = (keys, initialConfig, prefix, { removePrefixes }) => {
+    const re = new RegExp(`^${prefix}`, 'i');
+
+    const config = keys.reduce((acc, key) => {
+        if (re.test(key)) {
+            const name = removePrefixes ? key.replace(prefix, '') : key;
+
+            return {
+                ...acc,
+                [name]: process.env[key],
+            };
+        }
+
+        return acc;
+    }, initialConfig);
+
+    return config;
+};
+
+/**
+ * NextJS plugin to pass environment variables to NextJS's runtime configuration (`publicRuntimeConfig` and `serverRuntimeConfig`)
+ *
+ * @param {Object} options
+ * @param {String} options.publicPrefix     Prefix of variables to lookup and then pass to publicRuntimeConfig
+ * @param {String} options.serverPrefix     Prefix of variables to lookup and then pass to serverRuntimeConfig
+ * @param {String} options.removePrefixes   Option to remove prefix when passing variables to runtime config
+ */
+
+const runtimeEnv = (options = {}) => (nextConfig = {}) => {
```

Comment:
Do we need the default value for `nextConfig`? I think next always passes us a plain-object.

Reply:
It depends (see discussion above).

If we're using like this:

```js
module.exports = withRuntimeEnv()();
```

Then the `nextConfig` is `undefined`. I believe it's best to have this default value because of this example (same thing happens in all of **zeit**'s own plugins).